### DISCUSSION TITLE
Specifying --python-version in rez-pip should not default to venv pip

### DISCUSSION THREAD

**Discussion by hughetop** [AcademySoftwareFoundation/rez#1412] [General]
If I'm trying to install some 3rd-party pypi package with rez-pip and specify a python version, it should use that python version or error. Currently it will [fallback](https://github.com/AcademySoftwareFoundation/rez/blob/c0cc4bf1542a6d7a7b41fe5871e28743c1cd0f6b/src/rez/pip.py#L109) to pip installed into rez's virtual env, which is likely the wrong python version. This caused a lot of headache as I was trying to build packages for python 3.10, but it was silently failing and defaulting to 3.9. In my opinion, if you specify a python version and rez can't find that version's pip it should raise an error. If you aren't paying close attention, you would think it worked, then spend time trying to figure out why the package is missing.

I know this would be an API-breaking change, unless a new argument is added. Something like `--strict` could possibly make it so that if the specified python or pip version isn't found, it'll error instead of falling back.

**Comment by JeanChristopheMorinPerso** [comment]
I'd have to look back at our previous discussions on this, but I kind of remember that it was a deliberate choice. But I might be mistaken. Anyway, if we were to change the behavior, I don't think it would be a breaking change (at least I wouldn't consider that a breaking change).

**Reply by hughetop** [reply]
I can understand it being by design, especially the pip version, which is why I posted in discussions instead of issues. The thing that threw me off was that it _appeared_ to work until I looked closer at the log. Then I realized it was grabbing the wrong python version, even though I specified a different version. My suggestion would be to enforce the python version _only if `--python-version` is specified_, otherwise the default behavior works fine.

I can see both arguments for and against it being a breaking change, I have no strong opinions there. But an additional arg or a new key in the rezconfig would make it completely backwards-compatible.

**Reply by JeanChristopheMorinPerso** [reply]
I did some research. https://github.com/AcademySoftwareFoundation/rez/pull/667 contains some history around that. But most discussions where about how to find pip and not how to find Python. I think that the issue you reported might just be an oversight from us. And like I said, changing that wouldn't be considered a breaking change.

**Reply by hughetop** [reply]
Interesting, well this only came up due to a different, but related issue. I was able to work around that though.
